lib/query/querybuilder.js

```diff
@@ -441,6 +441,11 @@ class Builder extends EventEmitter {
       }
     }
 
+    assert(
+      !isObject(value),
+      'The values in where clause must not be object or array.'
+    );
```

Comment:
boolean value `false` , if not passed as a string, yields the same result (talking about MySQL instead of knex here) https://www.db-fiddle.com/f/w8CJ5SAYEKwJqAXyoYPSwC/2